### PR TITLE
Add Chapter 4

### PR DESCRIPTION
**Pull request by remo5000**
### Features
- Added some functions from the docs
- Imported some type checking functions from a previous repository
- Used external `parse` function from sourceacademy

### TODO
- [x] Add a `parse` function
- [x] Add the source for the `parse` function

Changing the grammar of the metacircular parser will be raised as an issue (#24 )

### CODE REVIEW

**Comment by ningyuansg** [comment]
@remo5000 You gotta fix the tests

**Comment by remo5000** [comment]
Tried a fix, let's see if it works for coveralls

**Comment by ningyuansg** [comment]
You've tested the the js files from this new setting in 10e1fa9 works on frontend?

File: scripts/build.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Compile all .ts files
+tsc
```
Comment (ningyuansg) [inline scripts/build.sh]: Do a `&&` so `cp` does not run if `tsc` fails.

**Comment by ningyuansg** [comment]
Bump

**Comment by remo5000** [comment]
Missed the notif, done! @ningyuansg

**Comment by martin-henz** [comment]
I think some more work is needed for this. The new specs for Source §4 are here. 
[source_4.pdf](https://github.com/source-academy/js-slang/files/2431261/source_4.pdf)

According to Yuchuan, apply_in_underlying_javascript is missing from the implementation, see page 7 of specs.

**Pull request merged**